### PR TITLE
[BUGFIX] Fix missing frontend.typoscript request attribute

### PR DESCRIPTION
**Pull request by tillhoerner**
Fixes: #3751

### CODE REVIEW

**Comment by dkd-kaehm** [comment]
@tillhoerner 
Thans for fixing that issue.

File: Classes/FrontendEnvironment/Tsfe.php [hunk 1]
```diff
@@ -168,10 +168,10 @@ protected function initializeTsfe(int $pageId, int $language = 0, ?int $rootPage
             $backedUpBackendUser = $GLOBALS['BE_USER'] ?? null;
             try {
                 $tsfe->determineId($serverRequest);
-                $serverRequest->withAttribute('frontend.controller', $tsfe);
+                $serverRequest = $serverRequest->withAttribute('frontend.controller', $tsfe);
```
Comment (dkd-kaehm) [inline Classes/FrontendEnvironment/Tsfe.php:171]: Makes it sense to move this line above the `$tsfe->determineId($serverRequest);`?
Like?
```suggestion
                $serverRequest = $serverRequest->withAttribute('frontend.controller', $tsfe);
                $tsfe->determineId($serverRequest);
```

Comment (dkd-kaehm) [inline Classes/FrontendEnvironment/Tsfe.php:171]: Maybe we should switch the lines, to provide the same instance of `$serverRequest`?

Comment (tillhoerner) [reply Classes/FrontendEnvironment/Tsfe.php:171]: Not really sure if that's needed. In the [`TypoScriptFrontendInitialization`](https://github.com/TYPO3/typo3/blob/12.4/typo3/sysext/frontend/Classes/Middleware/TypoScriptFrontendInitialization.php#L92) Middleware `determineId()` is also called before adding the `frontend.controller` attribute to the request.

**Review by dkd-kaehm — APPROVED** [review]
LGTM
:+1:

**Pull request merged**